static/js/controllers/confirm-modal.js

```diff
@@ -81,21 +90,13 @@
           if (result.then && typeof result.then === 'function') {
             // It's a promise!
             result
-              .then(function() {
-                $scope.ok();
-              })
-              .catch(function(err) {
-                _setErrorMode(err);
-              });
+              .then($scope.ok)
+              .catch(_setErrorMode);
             return;
           }
 
           // Simple true/false result.
-          if (result) {
-            $scope.ok();
-          } else {
-            _setErrorMode();
-          }
+          $scope.ok();
```

Comment:
Right, but in line 85 (of the changed version) we've already dealt with the processing function being synchronous and returning false. I can't see any code that would change the value of result between that line and here. Maybe I'm missing something.